src/odemis/acq/stream/_static.py

```diff
@@ -46,28 +46,82 @@ class StaticStream(Stream):
     def __init__(self, name, raw):
         """
         Note: parameters are different from the base class.
-        image (DataArray): static raw data.
-        raw (None or list of DataArrays): raw data to be used at initialisation
-          by default, it will contain no data.(None or list of DataArrays)
-        """
+        raw (list of DataArray or DataArrayShadow): The data to display.
+        """
+        if isinstance(raw, model.DataArray):
+            raw = [raw]
+        elif isinstance(raw, list):
+            if len(raw) > 0 and not isinstance(raw[0], model.DataArray):
+                raise ValueError("'raw' must be a list of DataArray")
+        elif isinstance(raw, model.DataArrayShadow):
+            if hasattr(raw, 'maxzoom'):
+                md = raw.metadata
+                # get the pixel size of the full image
+                # TODO check if an exception must be raised here
+                ps = md.get(model.MD_PIXEL_SIZE, (1e-6, 1e-6))
+                default_mpp = ps[0] * raw.maxzoom ** 2
+                # sets the mpp as the X axis of the pixel size of the full image
+                self.mpp = model.FloatVA(default_mpp, setter=self._mpp_setter)
+
+                full_rect = Stream._fullRect(raw)
+                self.rect = model.VigilantAttribute(full_rect, setter=self._rect_setter)
```

Comment:
rect must be a TupleContinuous (VA). This way, it will have a range. The advantages are that: it's automatically check the value given are correct (so you can remove the check in the setter), and it'll be possible of the GUI to know the full bounding box of the image. range is a tuple of tuples: 
```
l,t,r,b = full_rect 
rng = ((l,t,l,t), (r,b,r,b))
```